variantlib/loader.py

```diff
@@ -127,6 +127,49 @@ def get_supported_configs(cls) -> dict[str, ProviderConfig]:
 
         return provider_cfgs
 
+    @classmethod
+    @VariantCache()
+    def get_all_configs(cls) -> dict[str, ProviderConfig]:
+        """Get a mapping of namespaces to all valid configs"""
+
+        provider_cfgs = {}
+        for namespace, plugin_instance in cls._plugins.items():
+            key_configs = plugin_instance.get_all_configs()
+
+            if not isinstance(key_configs, list):
+                logging.error(
+                    "Provider: %(namespace)s returned an unexpected type: "
+                    "%(type)s - Expected: `list[KeyConfig]`. Ignoring...",
+                    {"namespace": namespace, "type": type(key_configs)},
+                )
+                continue
+
+            if not key_configs:
+                logging.error(
+                    "Provider: %(namespace)s does not support any configs!",
+                    {"namespace": namespace},
+                )
+                continue
+
+            for key_cfg in key_configs:
+                if not isinstance(key_cfg, KeyConfigType):
+                    logging.error(
```

Comment:
Please change to a complete exception raise. No "log and ignore" inside `variantlib`.
If there is a single error or anything unexpected: Exception => Variant behavior deactivated